base-template.sh

```diff
@@ -378,29 +395,30 @@ process_shared_hooks() {
 #####################################################
 update_shared_hooks_if_appropriate() {
     # run an init/update if we are after a "git pull" or triggered manually
-    if [ "$HOOK_NAME" = "post-merge" ] || [ "$HOOK_NAME" = ".githooks.shared.trigger" ]; then
+    if [ "$HOOK_NAME" = "post-merge" ] ||
```

Comment:
I don't really trust `A || B && C || D` in POSIX shell. Can you please extract them?
Maybe something like:
```
RUN_UPDATE="false"
[ "$HOOK_NAME" = "post-merge" ] && RUN_UPDATE="true"
[ "$HOOK_NAME" = ".githooks.shared.trigger" ] && RUN_UPDATE="true"
[ "$HOOK_NAME" = "post-checkout" ] && [ "$1" = "$GIT_NULL_REF" ] && RUN_UPDATE="true"
if [ "$RUN_UPDATE" = "true" ]; then
...
```